### DISCUSSION TITLE
Azure Pipeline failing (Could not find a schema.prisma file that is required for this command)

### DISCUSSION THREAD

**Discussion by KiriLucas** [prisma/prisma#11505] [Q&A] [answered]
Hey everyone!
I've got a NestJS application with a pipeline fully configured on Azure DevOps, now, I'm not very _DevOps fluent_ , so I may not be able to give you a fully correct explanation on my issue, but basically, my pipeline is failing because of Prisma (to be more specific, it's on the _Docker image building stage_ when I try to run `npx prisma migrate deploy` and `npx prisma generate`):

```console
You can either provide it with --schema, set it as `prisma.schema` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location

The command '/bin/sh -c npx prisma migrate deploy' returned a non-zero code: 1
##[error]The command '/bin/sh -c npx prisma migrate deploy' returned a non-zero code: 1
##[error]The process '/usr/bin/docker' failed with exit code 1
```

Because I'm using multi-stage building on my Dockerfile, I'm trying to run `prisma migrate` and `prisma generate` alongside my other dependencies handling, like this:

```Dockerfile
FROM node:16-alpine as dependencies

COPY ./package.json ./package-lock.json ./.npmrc /app/

RUN cd /app && npm ci --production
RUN npx prisma migrate deploy
RUN npx prisma generate
```

I have a prisma directory on my project's root with a `schema.prisma` file on it and it works like a charm when running it locally. Altough it's not an Azure Function, I used the following guide: [Prisma | Deploying to Azure Functions
](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-azure-functions
)

My issue might actually be because of my lack of knowledge on prisma and devops-related stuff in general, but in any case, I would appreciate if someone could point me to the right direction.

**Comment by janpio** [comment, marked as answer]
Can you please edit your message to include the full error message? This looks like only the second half of it (first half might be in the title - but better make sure).

**Reply by KiriLucas** [reply]
My bad, I actually forgot to reply to you!
Turns out my issue was caused by that `COPY` command on my original question which was not including `/prisma` directory to it.
To fix it, I just had to add it like this: 

```Dockerfile
COPY ./prisma ./package.json ./package-lock.json ./.npmrc /app/
```

I'm sorry again for taking so long to reply as you tried to help me so promptly! But that's it, it's working now :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
